components/servo_tracing/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+                    ));
+                },
+                Meta::NameValue(meta_name_value)
+                    if meta_name_value.path.is_ident("level") && level.is_none() =>
+                {
+                    level.replace(meta_name_value.value);
+                },
+                Meta::NameValue(meta_name_value)
+                    if meta_name_value.path.is_ident("level") && level.is_some() =>
+                {
+                    return syn::Result::Err(syn::Error::new(
+                        span,
+                        "Attribute argument `error` has been provided twice",
+                    ));
+                },
+                Meta::NameValue(meta_name_value)
+                    if meta_name_value.path.is_ident("name") && name.is_none() =>
+                {
+                    name.replace(meta_name_value);
+                },
+
+                Meta::NameValue(meta_name_value)
+                    if meta_name_value.path.is_ident("name") && name.is_some() =>
+                {
+                    return syn::Result::Err(syn::Error::new(
+                        span,
+                        "Attribute argument `name` has been provided more than once",
+                    ));
+                },
+                arg @ _ => {
+                    eprintln!("{:?}", arg.path().to_token_stream().to_string());
+                    passthrough_args.push(arg);
+                },
+            }
+        }
+
+        Ok(InstrumentConfiguration {
+            passthrough_args,
+            field_args,
+            level,
+            name,
+        })
+    }
+}
+
+/// injects servo_profiling as needed
+fn inject_servo_profiling(items: MetaList) -> syn::Result<MetaList> {
+    let metas: Punctuated<Meta, Comma> = Punctuated::parse_terminated.parse2(items.tokens)?;
+
+    let mut servo_profiling_given = false;
+    let mut empty = true;
+
+    for arg in metas.iter() {
+        servo_profiling_given |= arg.path().is_ident("servo_profiling");
+        empty = false;
+    }
+
+    let items = match (servo_profiling_given, empty) {
+        (true, _) => parse_quote! { fields(#metas) },
+        (false, true) => parse_quote! {
+            fields(servo_profiling = true)
+        },
+        (false, false) => {
+            parse_quote! {
+                fields(#metas, servo_profiling = true)
+            }
+        },
+    };
+
+    Ok(items)
+}
+
+fn instrument_internal(
+    attr: proc_macro2::TokenStream,
+    item: proc_macro2::TokenStream,
+) -> syn::Result<proc_macro2::TokenStream> {
+    dbg!(&attr);
+    // Prepare passthrough arguments for tracing::instrument
+    let InstrumentConfiguration {
+        mut passthrough_args,
+        field_args,
+        level,
+        name,
+    } = parse2(attr)?;
+    let input_fn: ItemFn = parse2(item)?;
+
+    // setup invariants to make reconstruction easier
+    let field_args = field_args.unwrap_or_else(|| parse_quote! { fields(servo_profiling = true) });
```

Comment:
I think we would want to inject `servo_profiling = true` into fields, if it is not there. I believe it was added so that we can easily filter out all tracing events that are not from servo but from dependencies, which also happen to be using tracing.

Reply:
We do that in `inject_servo_profiling`.